DAC/src/ringbuffer.c

```diff
@@ -0,0 +1,65 @@
+/*
+ * ringbuffer.c
+ *
+ *  Created on: Feb 6, 2018
+ *      Author: sharana
+ *      Description: Program to initiate a ring buffer.
+ */
+#include"ringbuffer.h"
+#include<inttypes.h>
+/*
+ * Here we have a method put characters into the buffer.
+ * @param buffer it is a structure type to access the correct ring buffer.
+ * @param element it is the character.
+ */
+void put(RingBuffer* buffer, char element){
+	//Check space
+	 if(hasSpace(&buffer)== 1)
+	    {	//adding element to the buffer
+	        buffer->buffer[buffer->put] = element;
+	        buffer->put = (buffer->put + 1) % BUF_SIZE;
+	    }else{
+	    	//it it is at the get
```

Comment:
Careful. This code removes something from the head of the ring buffer without adding anything to the tail of the ring buffer. Not putting is not what the put function should do.